tsconfig.eslint.json

```diff
@@ -1,6 +1,8 @@
 {
   "compilerOptions": {
-    "types": ["@types/node"]
+    "types": ["@types/node"],
+    "noEmit": true,
```

Comment:
you shouldn't need this as we add it automatically:
https://github.com/typescript-eslint/typescript-eslint/blob/ced9b26ce088eb4b3a39ddb0c5342373725c9912/packages/typescript-estree/src/create-program/useProvidedProgram.ts#L58

https://github.com/typescript-eslint/typescript-eslint/blob/ced9b26ce088eb4b3a39ddb0c5342373725c9912/packages/typescript-estree/src/create-program/shared.ts#L14-L22

Reply:
Yeah but those are invoked when we provide our own programs up front with the utility from parser, so I thought it made sense to keep the `tsconfig.eslint.json` in a state where it is independently working

Reply:
Oh sorry, you are just focusing on noEmit - you are right there!

Reply:
I was thinking of allowJs